EssentialCSharp.Web.Tests/FunctionalTests.cs

```diff
@@ -20,6 +20,44 @@ public async Task WhenTheApplicationStarts_ItCanLoadLoadPages(string relativeUrl
         Assert.Equal(HttpStatusCode.OK, response.StatusCode);
     }
 
+    [Theory]
+    [InlineData("/guidelines?rid=test-referral-id")]
+    [InlineData("/about?rid=abc123")]
+    [InlineData("/hello-world?rid=user-referral")]
+    public async Task WhenPagesAreAccessedWithRidParameter_TheyReturnContentSuccessfully(string relativeUrl)
+    {
+        using WebApplicationFactory factory = new();
+
+        HttpClient client = factory.CreateClient();
+        using HttpResponseMessage response = await client.GetAsync(relativeUrl);
+
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        
+        // Ensure the response has content (not blank)
+        string content = await response.Content.ReadAsStringAsync();
+        Assert.NotEmpty(content);
+        
+        // Verify it's actually HTML content, not just whitespace
+        Assert.Contains("<html", content, StringComparison.OrdinalIgnoreCase);
+    }
+
+    [Theory]
+    [InlineData("/guidelines?rid=")]
+    [InlineData("/about?rid=   ")]
+    public async Task WhenPagesAreAccessedWithEmptyRidParameter_TheyStillWork(string relativeUrl)
+    {
+        using WebApplicationFactory factory = new();
+
+        HttpClient client = factory.CreateClient();
+        using HttpResponseMessage response = await client.GetAsync(relativeUrl);
+
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        
+        string content = await response.Content.ReadAsStringAsync();
+        Assert.NotEmpty(content);
+        Assert.Contains("<html", content, StringComparison.OrdinalIgnoreCase);
+    }
```

Comment:
This is fine, but it seems like we could combine these into a single `WhenPagesAreAccessed_TheyReturnHtml`.

Same for the `WhenPagesAreAccessedWithNonRidParameters_TheyStillWork` below